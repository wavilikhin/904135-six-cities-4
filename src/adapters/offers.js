export const createOffer = (offer) => ({
  id: offer.id,
  city: offer.city.name,
  cityZoom: offer.city.location.zoom,
  isPremium: offer.is_premium,
  cityCoords: [offer.city.location.latitude, offer.city.location.longitude],
  image: offer.preview_image,
  priceValue: offer.price,
  name: offer.title,
  type: offer.type,
  coords: [offer.location.latitude, offer.location.longitude],
  bedrooms: offer.bedrooms,
  description: offer.description,
  goods: offer.goods,
  host: {
    avatar_url: offer.host.avatar_url,
    id: offer.host.id,
    isPro: offer.host.is_pro,
    name: offer.host.name,
  },
  images: offer.images,
  isFavorite: offer.is_favorite,
  location: {
    latitude: offer.location.latitude,
    longitude: offer.location.longitude,
    zoom: offer.location.zoom,
  },
  maxAdults: offer.max_adults,
  rating: offer.rating,
});
