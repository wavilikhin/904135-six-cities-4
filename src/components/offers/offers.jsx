import React from 'react';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import OffersList from '../offers-list/offers-list.jsx';
import Map from '../map/map.jsx';

const Offers = (props) => {
  const { offersDataArray, city } = props;

  const filtredOffers = offersDataArray.filter((offer) => {
    return offer.city === city;
  });

  return (
    <div className="cities">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">
            {filtredOffers.length} places to stay in {city}
          </b>
          <form className="places__sorting" action="#" method="get">
            <span className="places__sorting-caption">Sort by</span>
            <span className="places__sorting-type" tabIndex="0">
              Popular
              <svg className="places__sorting-arrow" width="7" height="4">
                <use xlinkHref="#icon-arrow-select"></use>
              </svg>
            </span>
            <ul className="places__options places__options--custom places__options--opened">
              <li
                className="places__option places__option--active"
                tabIndex="0"
              >
                Popular
              </li>
              <li className="places__option" tabIndex="0">
                Price: low to high
              </li>
              <li className="places__option" tabIndex="0">
                Price: high to low
              </li>
              <li className="places__option" tabIndex="0">
                Top rated first
              </li>
            </ul>
          </form>
          <OffersList />
        </section>
        <div className="cities__right-section">
          <section className="cities__map map">
            <Map city={city} zoom={12} offers={filtredOffers} />
          </section>
        </div>
      </div>
    </div>
  );
};

Offers.propTypes = {
  offersDataArray: PropTypes.arrayOf(
    PropTypes.shape({
      city: PropTypes.string.isRequired,
      cityCoords: PropTypes.arrayOf(PropTypes.number, PropTypes.number),
      quality: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      priceValue: PropTypes.string.isRequired,
      priceText: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
      coords: PropTypes.arrayOf(PropTypes.number, PropTypes.number).isRequired,
    }),
  ).isRequired,
  city: PropTypes.string,
};

const mapStateToProps = (state) => ({
  city: state.city,
  offersDataArray: state.offers,
});

export { Offers };
export default connect(mapStateToProps, null)(Offers);
