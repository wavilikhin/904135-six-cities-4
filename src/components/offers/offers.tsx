import * as React from 'react';
import { connect } from 'react-redux';
import OffersList from '../offers-list/offers-list';
import Map from '../map/map';
import { getCity, getFiltredOffers } from '../../reducer/data/selectors';
import MainEmpty from '../main-empty/main-empty';
import PlaceSorting from '../place-sorting/place-sorting';
import { OfferInfo } from '../../types';

interface Props {
  filtredOffers: OfferInfo[];
  city: string;
}

const Offers: React.FC<Props> = ({ filtredOffers, city }) => {
  return (
    <div className="cities">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">
            {filtredOffers.length} places to stay in {city}
          </b>
          <PlaceSorting />
          {filtredOffers.length === 0 ? (
            <MainEmpty city={city} />
          ) : (
            <OffersList />
          )}
        </section>
        <div className="cities__right-section">
          <section className="cities__map map">
            <Map offers={filtredOffers} />
          </section>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  city: getCity(state),
  filtredOffers: getFiltredOffers(state),
});

export { Offers };
export default connect(mapStateToProps, null)(Offers);
