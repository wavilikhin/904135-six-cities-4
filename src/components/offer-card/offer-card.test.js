import React from 'react';
import renderer from 'react-test-renderer';
import OfferCard from './offer-card.jsx';
import { Router } from 'react-router-dom';
import history from '../../history.js';
import { OFFERS } from '../../test/__mocks__/offers.js';

const cardData = OFFERS[0];

describe(`OfferCard component snapshot test`, () => {
  it(`Should render content card with given data`, () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <OfferCard
            cardData={cardData}
            userFavorites={[]}
            handleFavoritesUpdate={() => {}}
          />
        </Router>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
