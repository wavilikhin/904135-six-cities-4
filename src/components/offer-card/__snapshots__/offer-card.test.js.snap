// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard component snapshot test Should render content card with given data 1`] = `
<article
  className="cities__place-card place-card"
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          813
        </b>
        <span
          className="place-card__price-text"
        >
           
          ∕ night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button "
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        Wood and stone place
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      house
    </p>
  </div>
</article>
`;
